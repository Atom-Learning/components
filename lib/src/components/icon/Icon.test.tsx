import { Ok } from '@atom-learning/icons'
import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { Icon } from '.'

describe(`Icon component`, () => {
  it('renders an icon', async () => {
    const { container } = render(<Icon is={Ok} title="check" />)

    await screen.getByTitle('check')

    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(<Icon is={Ok} title="check" />)

    expect(await axe(container)).toHaveNoViolations()
  })
})
