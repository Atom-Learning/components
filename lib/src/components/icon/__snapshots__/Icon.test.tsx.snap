// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component renders an icon 1`] = `
@media  {
  .c-dbrbZt {
    display: inline-block;
    fill: none;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }
}

@media  {
  .c-dbrbZt-dMrjnF-size-md {
    height: var(--sizes-2);
    width: var(--sizes-2);
    stroke-width: 1.75;
  }
}

<div>
  <svg
    aria-hidden="true"
    class="c-dbrbZt c-dbrbZt-dMrjnF-size-md"
    title="check"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <polyline
      points="4 13 9 18 20 7"
    />
  </svg>
</div>
`;
